client/extensions/woocommerce/app/orders/orders-list.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * External dependencies
+ */
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import React, { Component } from 'react';
+import { times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import EmptyContent from 'components/empty-content';
+import { fetchOrders } from 'woocommerce/state/sites/orders/actions';
+import {
+	areOrdersLoading,
+	areOrdersLoaded,
+	getOrders,
+	getTotalOrdersPages
+} from 'woocommerce/state/sites/orders/selectors';
+import { getOrdersCurrentPage } from 'woocommerce/state/ui/orders/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteAdminUrl } from 'state/sites/selectors';
+import { setCurrentPage } from 'woocommerce/state/ui/orders/actions';
+import NavItem from 'components/section-nav/item';
+import NavTabs from 'components/section-nav/tabs';
+import SectionNav from 'components/section-nav';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+class Orders extends Component {
+	componentDidMount() {
+		const { siteId, currentPage } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchOrders( siteId, currentPage );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.currentPage !== this.props.currentPage || newProps.siteId !== this.props.siteId ) {
+			this.props.fetchOrders( newProps.siteId, newProps.currentPage );
+		}
+	}
+
+	getOrderStatus = ( status ) => {
+		const { translate } = this.props;
+		const classes = `orders__item-status is-${ status }`;
+		let paymentLabel;
+		let shippingLabel;
+		switch ( status ) {
+			case 'pending':
+				shippingLabel = translate( 'New order' );
+				paymentLabel = translate( 'Payment pending' );
+				break;
+			case 'processing':
+				shippingLabel = translate( 'New order' );
+				paymentLabel = translate( 'Paid in full' );
+				break;
+			case 'on-hold':
```

Comment:
Ah, now I see why we were getting two labels for some orders. Nevermind my other comment :P